src/components/Engines/Engines.tsx

```diff
@@ -1,73 +1,51 @@
-import React, { Component, ReactElement } from 'react';
+import React, { useContext } from 'react';
 import Grid from '@material-ui/core/Grid';
 import ListItemText from '@material-ui/core/ListItemText';
 
-import { VersionPageConsumerProps, DetailContextConsumer } from '../../pages/Version';
+import { DetailContext } from '../../pages/Version';
 import Avatar from '../../muiComponents/Avatar';
 import List from '../../muiComponents/List';
 import npm from '../Install/img/npm.svg';
 
 import { StyledText, EngineListItem } from './styles';
-// @ts-ignore
 import node from './img/node.png';
 
-const ICONS = {
-  'node-JS': <Avatar src={node} />,
-  'NPM-version': <Avatar src={npm} />,
-};
-
-class Engine extends Component {
-  public render(): ReactElement<HTMLElement> {
-    return (
-      <DetailContextConsumer>
-        {context => {
-          return this.renderEngine(context as VersionPageConsumerProps);
-        }}
-      </DetailContextConsumer>
-    );
-  }
-
-  private renderEngine = ({ packageMeta }): ReactElement<HTMLElement> | null => {
-    const { engines } = packageMeta.latest;
-    if (!engines) {
-      return null;
-    }
+const Engine: React.FC = () => {
+  const { packageMeta } = useContext(DetailContext);
 
-    const engineDict = {
-      'node-JS': engines.node,
-      'NPM-version': engines.npm,
-    };
+  const engines = packageMeta && packageMeta.latest && packageMeta.latest.engines;
```

Comment:
I saw that TypeScript 3.7 is specified in `package.json`, but `?.` doesn't seem to be supported by the tooling yet :(

Reply:
That's odd, I did not tried yet. I'll double check.

Reply:
I think it was prettier that was giving me issues, 1.19 with support for 3.7 isn't released yet: https://github.com/prettier/prettier/blob/master/CHANGELOG.unreleased.md

Reply:
Really? 😞 I didn't test the new features yet, but I thought we could already use it ... so let's update prettier :D